presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizerInformationCollector.java

```diff
@@ -14,21 +14,28 @@
 package com.facebook.presto.sql.planner.optimizations;
 
 import com.facebook.presto.spi.eventlistener.PlanOptimizerInformation;
+import com.google.common.collect.ImmutableList;
 
-import java.util.LinkedList;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Set;
 
+@ThreadSafe
 public class OptimizerInformationCollector
 {
-    private final List<PlanOptimizerInformation> optimizationInfo = new LinkedList<PlanOptimizerInformation>();
+    @GuardedBy("this")
+    private final Set<PlanOptimizerInformation> optimizationInfo = new LinkedHashSet<PlanOptimizerInformation>();
 
-    public void addInformation(PlanOptimizerInformation optimizerInformation)
+    public synchronized void addInformation(PlanOptimizerInformation optimizerInformation)
     {
         this.optimizationInfo.add(optimizerInformation);
     }
 
-    public List<PlanOptimizerInformation> getOptimizationInfo()
+    public synchronized List<PlanOptimizerInformation> getOptimizationInfo()
```

Comment:
return a set

Reply:
I was intentionally using an LInkedHashSet hoping to preserve the order of which the rules are triggered. If we return a 
copy of Immutable set here, will the order be lost?

Reply:
I think immutable set maintains insertion order https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableSet.html#copyOf(java.util.Collection)